spec.md

```diff
@@ -582,6 +582,20 @@ message ControllerPublishVolumeRequest {
   // Whether to publish the volume in readonly mode. This field is
   // REQUIRED.
   bool readonly = 5;
+
+  // OPTIONAL field for storing principal information to be used in
+  // this publishing of this volume. If the principal is present the
+  // plugin MAY ensure all access to the volumes contents through
+  // this mount is mapped to the identity specified by the principal.
+  // If no principal is provided the access ids are not mapped but handed
+  // through untouched.
+  // As this field is specific to a given publishing of a volume
+  // containers in a container hierarchy have to either have their own
+  // individual volume publishings or a parent container MAY provide
+  // access to his volume mount throught some CO specific internal
+  // mechanism. In the latter case still all access through that volume
+  // mount is mapped to the given principal, as in a standalone container.
+  Principal accessed_as = 6;
```

Comment:
@oritnm ah I see, so really this is about whether the stuff belongs in `ControllerPublishVolume` or `NodePublishVolume`?  To me it is a bit more flexible to put it in `ControllerPublishVolume` as the plugin might need it in order to do whatever work will happen there, and can propagate it to `NodePublishVolume` by putting it in `PublishVolumeInfo` in the response, which will get passed along to `NodePublishVolume`.  Having said that, I cannot think of an example off the top of my head where we would want container specific credentials in order to do something in `ControllerPublishVolume` so maybe that flexibility is unnecessary.